trailingstoploss_tp.py

```diff
@@ -108,39 +110,67 @@ def upgrade_config(thelogger, cfg):
                 cfg.write(cfgfile)
 
             thelogger.info("Upgraded section %s to have config list" % cfgsection)
-
+        if cfgsection.startswith("tsl_tp_") and cfg.has_option(cfgsection, "config"):
+            jsonconfiglist = list()
+            for configsectionconfig in json.loads(cfg.get(cfgsection, "config")):
+                if "sl-timeout" not in configsectionconfig:
+                    cfgsectionconfig = {
+                        "activation-percentage": configsectionconfig.get("activation-percentage"),
+                        "initial-stoploss-percentage": configsectionconfig.get("initial-stoploss-percentage"),
+                        "sl-timeout": "0",
+                        "sl-increment-factor": configsectionconfig.get("sl-increment-factor"),
+                        "tp-increment-factor": configsectionconfig.get("tp-increment-factor"),
+                    }
+                    jsonconfiglist.append(cfgsectionconfig)
+
+            if len(jsonconfiglist) > 0:
+                cfg.set(cfgsection, "config", json.dumps(jsonconfiglist))
+
+                with open(f"{datadir}/{program}.ini", "w+") as cfgfile:
+                    cfg.write(cfgfile)
+
+                thelogger.info("Upgraded section %s to have config list" % cfgsection)
     return cfg
 
 
-def update_deal(thebot, deal, new_stoploss, new_take_profit):
+def update_deal(thebot, deal, new_stoploss, new_take_profit, sl_timeout):
     """Update bot with new SL and TP."""
 
     bot_name = thebot["name"]
     deal_id = deal["id"]
 
+    payload = {
+        "deal_id": thebot["id"],
+        "stop_loss_percentage": new_stoploss,
+        "take_profit": new_take_profit,
+    }
+
+    # Check whether there is a timeout set for this config, when there is, set the timeout to this value.
+    # Otherwise, send the payload as it is.
+    if sl_timeout != 0:
+        payload["stop_loss_timeout_enabled"] = True
+        payload["stop_loss_timeout_in_seconds"] = sl_timeout
+
```

Comment:
Dit stukje gaat, denk ik, niet helemaal goed werken. Volgens mij moeten we altijd de sl-timeout overnemen die binnenkomt en die moet ook 0 kunnen zijn. Stel je eens het volgende voor; in de config heb ik twee profit configs. In de eerste heb ik een sl-timeout van 600 omdat ik verwacht dat de deal wat meer zal bouncen, in de tweede wil ik de sl-timeout op 0 hebben want dan wil ik gewoon sluiten. 

Zoals de code nu is, zal de 0 uit de tweede config nooit meer doorkomen en mijn sl-timeout dus altijd op 600 blijven staan in de deal.